onnx/defs/tensor/defs.cc

```diff
@@ -100,9 +100,30 @@ ONNX_OPERATOR_SET_SCHEMA(
     5,
     OpSchema()
         .SetDoc(Reshape_ver5_doc)
-        .Input(0, "data", "An input tensor.", "T")
-        .Input(1, "shape", "Specified shape for output.", "tensor(int64)")
-        .Output(0, "reshaped", "Reshaped data.", "T")
+        .Input(0,
+            "data",
+            "An input tensor.",
+            "T",
+            OpSchema::Single,
+            true,
+            1,
+            OpSchema::Differentiable)
```

Comment:
How to prove it?
1. Pytorch/Tensorflow has this differentiable.
2. Add math of backward to operator and ask reviewer to review the math.
3. Implement a backward using an existing auto-diff library.